exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/Metadata.java

```diff
@@ -1008,8 +1008,24 @@ public void setMax(Object max) {
       return nulls;
     }
 
-    @Override public boolean hasSingleValue() {
-      return (max != null && min != null && max.equals(min));
+    /**
+     * Checks that the column chunk has single value.
+     * Returns true if min and max are the same, but not null.
+     * Returns true if min and max are null and the number of null values
+     * in the column chunk is greater than 0.
+     *
+     * @return true if column has single value
```

Comment:
Thanks for such detailed analysis. This bug is also reproduced on master version, but it appears due to the wrong result of this method. Added a comparison of `nulls` with rows count to fix this bug and allow to return true for the case when field has only nulls.